homeassistant/components/aemet/__init__.py

```diff
@@ -0,0 +1,97 @@
+"""The AEMET OpenData component."""
+import asyncio
+import logging
+
+from aemet_opendata.interface import AEMET
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_API_KEY,
+    CONF_LATITUDE,
+    CONF_LONGITUDE,
+    CONF_MODE,
+    CONF_NAME,
+)
+from homeassistant.core import HomeAssistant
+
+from .const import (
+    COMPONENTS,
+    DOMAIN,
+    ENTRY_NAME,
+    ENTRY_WEATHER_COORDINATOR,
+    UPDATE_LISTENER,
+)
+from .weather_update_coordinator import WeatherUpdateCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup(hass: HomeAssistant, config: dict) -> bool:
+    """Set up the AEMET OpenData component."""
+    hass.data.setdefault(DOMAIN, {})
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, config_entry: ConfigEntry):
+    """Set up AEMET OpenData as config entry."""
+    name = config_entry.data[CONF_NAME]
+    api_key = config_entry.data[CONF_API_KEY]
+    latitude = config_entry.data.get(CONF_LATITUDE, hass.config.latitude)
+    longitude = config_entry.data.get(CONF_LONGITUDE, hass.config.longitude)
+    forecast_mode = _get_config_value(config_entry, CONF_MODE)
+
+    aemet = AEMET(api_key)
+    weather_coordinator = WeatherUpdateCoordinator(
+        aemet, latitude, longitude, forecast_mode, hass
+    )
+
+    await weather_coordinator.async_refresh()
+
+    hass.data.setdefault(DOMAIN, {})
```

Comment:
This is already done above.

Reply:
As I already told @MatthewFlamm, I based my integration on OpenWeatherMap, and this is inherited from there: https://github.com/home-assistant/core/blob/fd177441b3cfddb22b893b26e91862307bbcc437/homeassistant/components/openweathermap/__init__.py#L62

Reply:
Well we shouldn't add not needed code.